Service/Repository.php

```diff
@@ -206,6 +208,82 @@ public function execute(Search $search, $resultsType = Result::RESULTS_OBJECT)
         return $this->manager->execute([$this->type], $search, $resultsType);
     }
 
+
+    /**
+     * Parses scroll configuration from raw response.
+     *
+     * @param $raw
+     * @param null $scrollDuration
+     *
+     * @return array
+     */
+    private function getScrollConfiguration($raw, $scrollDuration = null)
+    {
+        $scrollConfig = [];
+        if (isset($raw['_scroll_id'])) {
+            $scrollConfig['_scroll_id'] = $raw['_scroll_id'];
+            $scrollConfig['duration'] = $scrollDuration;
+        }
+
+        return $scrollConfig;
+    }
+
+
+    /**
+     * Returns DocumentIterator with composed Document objects from array response.
+     *
+     * @param Search $search
+     *
+     * @return DocumentIterator
+     */
+    public function findDocument(Search $search)
```

Comment:
For documents it's pretty clear to understand what it means. But for arrays and raw below, I would suggest `findAsArray` and `findAsRaw`. Also `findAsDocuments` then. What do you think?